float64[] d_left
float64[] d_right
float64 init_s